coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -2,23 +2,28 @@
 
 
 @generate_repr()
-@generate_eq("documentation", "marker", "range")
+@generate_eq("documentation", "language", "docstyle", "marker", "range")
 class DocumentationComment:
     """
     The DocumentationComment holds information about a documentation comment
     inside source-code, like position etc.
     """
 
-    def __init__(self, documentation, marker, range):
+    def __init__(self, documentation, language,
+                 docstyle, marker=None, range=None):
```

Comment:
using `/* comment */` as documentation would be in fact not valid (it's technically valid, but semantically it isn't), because then the markers are somehow integrated into the documentation. The idea was to split this in this class already. Your parsing function would also read the implicit markers and treat them as actual documentation.

If people use this class, they need to deal with this kind of stuff. If they want to store documentation text, they can use strings :P

Otherwise this class is not really meaningful imo^^